FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download/DemoTasks/S3DownloadHTTPExample.c

```diff
@@ -700,14 +923,188 @@ static BaseType_t prvDownloadS3ObjectFile( const TransportInterface_t * pxTransp
      * headers is reused here. */
-    xResponse.pBuffer = ucUserBuffer;
-    xResponse.bufferLen = democonfigUSER_BUFFER_LENGTH;
+    pxResponse->pBuffer = ucUserBuffer;
+    pxResponse->bufferLen = democonfigUSER_BUFFER_LENGTH;
+
+    xHttpStatus = HTTPClient_InitializeRequestHeaders( &xRequestHeaders,
+                                                       &xRequestInfo );
+    if( xHttpStatus != HTTPSuccess )
+    {
+        LogError( ( "Failed initialize HTTP headers: Error=%s.",
+                    HTTPClient_strerror( xHttpStatus ) ) );
+        xStatus = pdFAIL;
+    }
+
+    if( xStatus == pdPASS )
+    {
+        /* Add the X-AMZ-DATE required headers to the request. */
+        xHttpStatus = HTTPClient_AddHeader( &xRequestHeaders,
+                                            SIGV4_HTTP_X_AMZ_DATE_HEADER,
+                                            sizeof( SIGV4_HTTP_X_AMZ_DATE_HEADER ) - 1,
+                                            cDateISO8601,
+                                            SIGV4_ISO_STRING_LEN );
+        if( xHttpStatus != HTTPSuccess )
+        {
+            LogError( ( "Failed to add X-AMZ-DATE to request headers: Error=%s.",
+                        HTTPClient_strerror( xHttpStatus ) ) );
+            xStatus = pdFAIL;
+        }
+    }
+
+    if( xStatus == pdPASS )
+    {
+        /* S3 requires the security token as part of the canonical headers. */
+        xHttpStatus = HTTPClient_AddHeader( &xRequestHeaders,
+                                            SIGV4_HTTP_X_AMZ_SECURITY_TOKEN_HEADER,
+                                            sizeof( SIGV4_HTTP_X_AMZ_SECURITY_TOKEN_HEADER ) - 1,
+                                            pcSecurityToken,
+                                            xSecurityTokenLen );
+        if( xHttpStatus != HTTPSuccess )
+        {
+            LogError( ( "Failed to add X-AMZ-SECURITY-TOKEN to request headers: Error=%s.",
+                        HTTPClient_strerror( xHttpStatus ) ) );
+            xStatus = pdFAIL;
+        }
+    }
+
+    if( xStatus == pdPASS )
+    {
+        /* Add Range header */
+        xHttpStatus = HTTPClient_AddRangeHeader( &xRequestHeaders,
+                                                 ulRangeStart,
+                                                 ulRangeEnd );
+        if( xHttpStatus != HTTPSuccess )
+        {
+            LogError( ( "Failed to add range to request headers: Error=%s.",
+                        HTTPClient_strerror( xHttpStatus ) ) );
+            xStatus = pdFAIL;
+        }
+    }
+
+    if( xStatus == pdPASS )
+    {
+        /* Add the SHA256 of an empty payload. */
+        xHttpStatus = HTTPClient_AddHeader( &xRequestHeaders,
+                                           SIGV4_HTTP_X_AMZ_CONTENT_SHA256_HEADER,
+                                           sizeof( SIGV4_HTTP_X_AMZ_CONTENT_SHA256_HEADER ) - 1,
+                                           S3_REQUEST_EMPTY_PAYLOAD_SHA256,
```

Comment:
I think its is better to compute the SHA256 payload in the demo instead of hardcoding it. In P2 we have computed it by passing the empty string([https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/main/demos/http/http_demo_s3_download/http_demo_s3_download.c#L1104](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/main/demos/http/http_demo_s3_download/http_demo_s3_download.c#L1104))